torch/distributed/_tensor/ops/pointwise_ops.py

```diff
@@ -376,11 +394,109 @@
+        followed_strategy = op_schema.kwargs_schema["out"]
+    else:
+        # normal pointwise op, we choose to follow the arg with
+        # the max shards in case operands needs reshard
+        for idx, arg_strategy in enumerate(op_schema.args_schema):
+            if not isinstance(arg_strategy, OpStrategy):
+                continue
+
+            arg_max_shards = arg_strategy.max_num_shards()
+            if arg_max_shards > max_shards:
+                max_shards_strategy_index = idx
+                max_shards = arg_max_shards
+
+        followed_strategy = op_schema.args_schema[max_shards_strategy_index]
+        assert isinstance(followed_strategy, OpStrategy)
+        follow_operand_dims_map = infer_broadcast_dims_map(
+            common_shape, followed_strategy.output_shape
+        )
+
+    assert isinstance(
+        followed_strategy, OpStrategy
+    ), f"no strategy to follow for {op_schema}!"
+    pointwise_strategy = OpStrategy([])
+
+    for placement_strategy in followed_strategy.strategies:
+        spec_to_follow = placement_strategy.output_spec
+        out_placements: List[Placement] = []
+
+        for placement in spec_to_follow.placements:
+            if isinstance(placement, Shard):
+                shard_dim = normalize_dim(placement.dim, len(spec_to_follow.shape))
+                common_ndim = len(common_shape)
+                new_shard_dim = common_ndim - len(spec_to_follow.shape) + shard_dim
+                out_placements.append(Shard(new_shard_dim))
+            else:
+                out_placements.append(placement)
+
+        input_specs = []
+        redistribute_costs: List[List[float]] = []
+        for idx, input_arg in enumerate(op_schema.args_schema):
+            if isinstance(input_arg, OpStrategy):
+                if idx == max_shards_strategy_index:
+                    # the current input arg is the one we want to follow
+                    input_specs.append(spec_to_follow)
+                    redistribute_costs.append([0] * len(input_arg.strategies))
+                    continue
+
+                # every arg follow the out_placements, but need to handle broadcasting
+                input_arg_spec = input_arg.strategies[0].output_spec
+                input_arg_dims_map = infer_broadcast_dims_map(
+                    common_shape, input_arg_spec.shape
+                )
+                input_placements = map_placements_after_broadcast(
+                    tuple(out_placements),
+                    common_shape,
+                    input_arg_dims_map,
+                )
+                input_specs.append(
+                    DTensorSpec(
+                        mesh=mesh,
+                        placements=input_placements,
+                        tensor_meta=input_arg_spec.tensor_meta,
+                    )
+                )
+                redistribute_costs.append(
+                    generate_redistribute_costs(input_arg, spec_to_follow)
```

Comment:
Wait, IIUC, you want to use `input_placements` to calculate `redistribute_cost`?

Reply:
Yeah good catch